runtime/compiler/optimizer/StringPeepholes.cpp

```diff
@@ -2092,9 +2143,25 @@ bool TR_StringPeepholes::checkMethodSignature(TR::SymbolReference *symRef, const
  */
 bool TR_StringPeepholes::skipNodeUnderOSR(TR::Node *node)
    {
-   return comp()->getOption(TR_EnableOSR)
-      && comp()->isOSRTransitionTarget(TR::postExecutionOSR)
-      && comp()->getMethodSymbol()->isOSRRelatedNode(node);
+   bool result = false;
+   if (comp()->getOption(TR_EnableOSR)
+      && comp()->isOSRTransitionTarget(TR::postExecutionOSR))
```

Comment:
I don't know why the transition target affects this any OSR helper for a pre or post transitionw ould need to be skipable.

Reply:
I think this needs a fast return for involuntary OSR where no skip is needed.